tools/makemanifest.py

```diff
@@ -173,15 +196,18 @@ def freeze_internal(kind, path, script, opt):
         manifest_list.append((kind, path, script, opt))
 
 def main():
-    global TOP
+    global TOP, LIB, PORT, BOARD
```

Comment:
I'd use the same names as the replacements, easier to read.
But see previous comment: I still think using argparse and perhaps arbitrary key-value pars is a better solution overall